pkgs/data/misc/sing-geoip/default.nix

```diff
@@ -0,0 +1,60 @@
+{ lib
+, stdenvNoCC
+, buildGoModule
+, fetchFromGitHub
+, clash-geoip
+}:
+
+let
+  generator = buildGoModule rec {
+    pname = "sing-geoip";
+    version = "unstable-2022-07-05";
+
+    src = fetchFromGitHub {
+      owner = "SagerNet";
+      repo = pname;
+      rev = "2ced72c94da4c9259c40353c375319d9d28a78f3";
+      hash = "sha256-z8aP+OfTuzQNwOT3EEnI9uze/vbHTJLEiCPqIrnNUHw=";
+    };
+
+    vendorHash = "sha256-lr0XMLFxJmLqIqCuGgmsFh324jmZVj71blmStMn41Rs=";
+
+    patchPhase = ''
+      # The codes args should start from the third args
```

Comment:
Why though?

Reply:
I thought this is a bug. The code take those args as a list and check the length of the list. If the list is empty, the data of all contries are writen to the file. Else only the specified contries are used. However, it puts the second arg, the output path, into the list so that the list is never empty.